backend/api_calls.py

```diff
@@ -577,7 +584,39 @@ def SendChatMessage(request, game_state):
   if not game_state.get('/groups/%s/players' % group, request['playerId']):
     raise InvalidInputError('You are not a member of that chat room.')
 
-  # TODO Scan message for any @all or @player to turn into notifications.
+  user_id = game_state.get('/playersPublic/%s' % request['playerId'], 'userId')
+  players_in_room = helpers.GetPlayerNamesInChatRoom(game_state, chat)
+  notification_data = {
+    'notificationId': 'notification-%s' % request['messageId'][len('message-'):],
+    'message': request['message'],
+    'app': True,
+    'vibrate': True,
+    'destination': 'TODO',
+    'sendTime': time.time(),
+    'icon': 'TODO'
+  }
+  # If we check for all @all, then there is no need to send out additional
+  # player notifications.
+  if '@all' in request['message'] and helpers.IsAdmin(game_state,
```

Comment:
Can we do a regex check for "@all\b", so we don't accidentally catch @allison or @allomancer?